test/test_jit.py

```diff
@@ -5595,7 +5599,20 @@ def test(op, const, swap_args):
             if op == '%' and swap_args is True:
                 continue
 
-            test(op, const, swap_args)
+            test(op, tensor, const, swap_args)
+
+        alpha_template = dedent('''
+        def func(t):
+            return torch.{{op}}({{lhs}}, {{rhs}}, alpha={alpha})
+        ''')
+        template = alpha_template.format(alpha="5.2")
+        test('add', float_tensor, 1, False, template=template)
+        test('add', long_tensor, 1.5, False, template=template)
+        test('sub', float_tensor, 1.5, False, template=template)
+        test('sub', float_tensor, 2, False, template=template)
+        template = alpha_template.format(alpha="5")
+        test('add', long_tensor, 1, False, template=template)
```

Comment:
These are just testing that the code compiles, and not any shape analysis properties

Reply:
isn't it verifying that the dtype returned by the jit compile/run is the same as the dtype returned when running the code normally?

Reply:
line 5577, `self.assertEqual(res1.dtype, res2.dtype, message + "\nres1=" + str(res1) + "\nres2=" + str(res2))` ?

Reply:
That's testing run in script vs run in python. Doesn't do anything with shape analysis

Reply:
ah... I thought running torch script was what relied on shape analysis. how can I test shape analysis then?

Reply:
I would probably do something like:

```
@torch.jit.script
def test():
    return torch.tensor(0.1)

out_val = test()
g = test.graph_for()
inferred_out_type = next(g.outputs()).type()
self.assertTrue(inferred_out_type.scalarType() = out_val.dtype)
self.assertTrue(inferred_out_type.dim() = out_val.dim)
```

You will have to test separately for the cases where we can't infer the output type (as with a NumberType input as i mentioned)